packages/bot-web-ui/src/pages/tutorials/guide-content/__tests__/guide-content.spec.tsx

```diff
@@ -47,21 +53,90 @@ describe('<FAQContent />', () => {
     });
     const mock_DBot_store = mockDBotStore(mock_store, mock_ws);
 
-    beforeEach(() => {
+    beforeAll(() => {
         wrapper = ({ children }: { children: JSX.Element }) => (
             <StoreProvider store={mock_store}>
                 <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
                     {children}
                 </DBotStoreProvider>
             </StoreProvider>
         );
-        render(<GuideContent guide_tab_content={mock_user_guide_content} video_tab_content={mock_guide_content} />, {
-            wrapper,
-        });
     });
 
-    it('should render the component', () => {
+    it('should render the GuideContent component', () => {
+        render(<GuideContent {...mocked_props} />, { wrapper });
+
         expect(screen.getByText('Step-by-step guides')).toBeInTheDocument();
         expect(screen.getByText('Videos on Deriv Bot')).toBeInTheDocument();
     });
+
+    it('should not define the active tour with function the setActiveTour() when it is the desktop version.', async () => {
+        mock_store.ui.is_desktop = true;
+        render(<GuideContent {...mocked_props} />, {
+            wrapper,
+        });
+
+        await waitFor(() => {
+            expect(mock_DBot_store.dashboard.active_tour).toBe('');
+        });
+    });
+
+    it('should open the dialog upon clicking the tutorial video button, should open the tab dashboard upon clicking tutorial card button.', async () => {
+        mock_store.ui.is_desktop = false;
+        const { container } = render(<GuideContent {...mocked_props} />, {
+            wrapper,
+        });
+
+        // eslint-disable-next-line testing-library/no-node-access, testing-library/no-container
+        const tutorial_video_button = container.querySelector('.tutorials-wrap__placeholder__button-group--play');
+        userEvent.click(tutorial_video_button);
+
+        await waitFor(() => {
+            expect(mock_DBot_store?.dashboard.is_dialog_open).toBeTruthy();
+        });
+
+        const tutorial_card_button = screen.getAllByTestId('tutorials-wrap--tour')[0];
+        userEvent.click(tutorial_card_button);
+
+        await waitFor(() => {
+            expect(mock_DBot_store?.dashboard.setActiveTab(DBOT_TABS.DASHBOARD));
+        });
+    });
+
+    it('should trigger the keyDown event upon clicking "Enter"', async () => {
+        const mockEventListener = jest.fn();
+        render(<GuideContent {...mocked_props} />, {
+            wrapper,
+        });
+
+        document.addEventListener('keydown', mockEventListener);
+        const button_cancel = screen.getAllByTestId('tutorials-wrap--tour')[1];
```

Comment:
@maryia-matskevich-deriv any reason why are we triggering enter keydown here and expecting the same keydown below ?

Reply:
we have a comment about this on the next line:
<img width="886" alt="Screenshot 2024-08-26 at 17 28 36" src="https://github.com/user-attachments/assets/235d1367-8cd5-46a0-bde6-c832ea63764d">

Reply:
if you have any better suggestions please share them😌

Reply:
<img width="985" alt="image" src="https://github.com/user-attachments/assets/7c2d30be-26cb-4d62-81af-6d73683b6ebe">
hi @maryia-matskevich-deriv i think these are the test cases without event listener.

Reply:
this also covers the missing branch 
![Uploading image.png…]()

Reply:
Also we just need to add a tab-index here
<img width="474" alt="image" src="https://github.com/user-attachments/assets/54fc3f6a-3155-49c7-a42b-6215a9920b81">